### PR TITLE
Ruby oop

### PR DESCRIPTION
**Pull request by redwing555**
This is the first milestone of the school library project it consists of : 

- Creating class `Person` with Instance vars: `@id`, `@name`, and `@age`.

1. Getters for `@id`, `@name`, and `@age`.
2. Setters for `@name` and `@age`.
3. Private method `of_age?`.
4. Public method `can_use_services?`

- Creating class` Student` with the following:

1.  Inherits from `Person`.
2.  Constructor extends parent's constructor by adding` @classroom` and a default parameter for it.
3.  Method `play_hooky` that returns "¯\(ツ)/¯".

- Creating class `Teacher` with the following:

1. Inherits from` Person`.
2. Constructor extends parent's constructor by adding `@specialization` and a default parameter for it.
3. Override `can_use_services?` so it always returns `true`.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = 111
```
Comment (henatan99) [inline person.rb]: - [x] You have done nice work overall. I think id should not be a constant value. You can use a random number generator here. Kindly make changes. :pray:

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @redwing555 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by er-contreras — COMMENTED** [review]
Hi @redwing555, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

wrong way

## Invalid Code Review Request

You have submitted a project, that has rubocop linter errors.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by redwing555** [comment]
@eri8-9 thank you for marking it as invalid, it is fixed now, happy coding cheers!

**Review by o-rashad — APPROVED** [review]
Asslam Alaikum @redwing555 ,😊

Hi👋🏽, Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by redwing555** [comment]
wa Alaykum Assalam @od-c0d3r 🤝  , thank you for the review,  cheers!

**Pull request merged**